File: javascripts/js/h00.overriding/Shape.js
/******* CLASS DEFINITION *******/
var Shape = (function () {
  var width, height;

  /******* CONSTRUCTOR *******/
  function Shape(width, height) {
    this.width = width;
    this.height = height;
  }

  /******* METHODS *******/
  Shape.prototype.calculateArea = function () { return 0.0; };

  return Shape;
})();

File: cs/i00.override/Shape.cs
public class Shape {
   /******* FIELDS *******/
   protected double width;
   protected double height;

   /******* CONSTRUCTORS *******/
   public Shape(double width, double height)  {
     this.width = width;
     this.height = height;
   }

   public Shape(int width, int height)  {
     this.width  = System.Convert.ToInt32(width);   // convert to double
     this.height = System.Convert.ToInt32(height);  // convert to double
   }

   /******* METHODS *******/
   public virtual double calculateArea() { return 0.0; }
}
